python/cudf/cudf/_lib/copying.pyx

```diff
@@ -190,60 +191,80 @@ def gather(
     return columns_from_unique_ptr(move(c_result))
 
 
-def scatter(object source, Column scatter_map, Column target_column,
-            bool bounds_check=True):
-    """
-    Scattering input into target as per the scatter map,
-    input can be a list of scalars or can be a table
-    """
-
-    cdef column_view scatter_map_view = scatter_map.view()
-    cdef table_view target_table_view = table_view_from_columns(
-        (target_column,))
-    cdef bool c_bounds_check = bounds_check
+cdef scatter_scalar(list source_device_slrs,
```

Comment:
Something to think about: is there a better way for us to handle scalar vs column functions? We have this division in a lot of our Cython layers, can we think of a way to reduce it? One option would be to define generic scalar/column functions (not sure exactly what that would look like) and then passing specialized functions as arguments to handle different features. This is a very speculative idea, so no need to have an answer in this PR, but I do dislike duplicating this scalar/column division over and over.